app/controllers/application_controller.rb

```diff
@@ -45,7 +45,7 @@ def after_sign_in_path_for(resource)
     set_flash_warning_reminder_to_update_details resource
     return edit_user_path id: current_user.id if session[:pending_organisation_id]
     return organisation_path(current_user.organisation) if current_user.organisation
-    return session[:previous_url] if session[:previous_url]
+    return session[:previous_url] if session[:previous_url] && session[:previous_url] != organisations_search_path
```

Comment:
so I might break this out into a separate method that explains what we are doing.  Exceeding the line length limit is often a hint that the line is going to be difficult to read, so how about:

```ruby
return session[:previous_url] if appropriate_previous_url_available?
```

```ruby
def appropriate_previous_url_available?
  session[:previous_url] && session[:previous_url] != organisations_search_path
end
```